packages/template-retail-react-app/app/components/island/index.jsx

```diff
@@ -84,7 +85,8 @@ function findChildren(children, componentType) {
  */
 function Island(props) {
     const {children} = props
-    if (!PARTIAL_HYDRATION_ENABLED) {
+    const isEnabled = getConfig()?.app?.partialHydrationEnabled ?? PARTIAL_HYDRATION_ENABLED // in a backward compatible way
```

Comment:
We also still support `PARTIAL_HYDRATION_ENABLED` for backward compatibility. But `config.app.partialHydrationEnabled` will take precedence if defined.